test/nodejs/reclinearregression.js

```diff
@@ -261,5 +358,60 @@ describe('RecursiveLinearRegression Tests', function () {
             assert.eqtol(linreg.weights.minus(linreg2.weights).norm(), 0, 1e-8);
         })
     });
+    describe('testing forget factor', function () {
+        it('should return correct weights', function () {
+            var regFact = 0.1;
+            var forgetFact = 0.90;
+            var n = 1000;
+            var dim = 3;
+
+            var linreg = new analytics.RecLinReg({ dim: dim, regFact: regFact, forgetFact: forgetFact });
+
+            var X = new la.Matrix({ rows: n, cols: dim });
+            var yArr = [];
+
+            for (var rowN = 0; rowN < n; rowN++) {
+                var rowArr = [1];
+                for (var colN = 1; colN < dim; colN++) {
+                    rowArr.push(Math.random());
+                }
+
+                var xi = new la.Vector(rowArr);
+                var yi = Math.random();
+
+                linreg.partialFit(xi, yi);
+
+                X.setRow(rowN, xi);
+                yArr.push(yi);
+            }
+
+            var y = new la.Vector(yArr);
 
-})
\ No newline at end of file
+            var wgts = linreg.weights;
+            var expectedWgts = linregWgtsFromMatrixForgetFact(X, y, regFact, forgetFact);
+
+            assert.eqtol(wgts[0], expectedWgts[0], tol);
+            assert.eqtol(wgts[1], expectedWgts[1], tol);
+            assert.eqtol(wgts[2], expectedWgts[2], tol);
+        })
+        it('should not become ill-formed', function () {
+            var regFact = 0.1;
+            var forgetFact = 0.5;
+            var n = 10000;
+            var dim = 3;
+
+            var ones = [];
+            for (var j = 0; j < dim; j++) {
+                ones.push(1);
+            }
+
+            var linreg = new analytics.RecLinReg({ dim: dim, regFact: regFact, forgetFact: forgetFact });
+
+            assert.doesNotThrow(function () {
+                for (var i = 0; i < n; i++) {
+                    linreg.partialFit(new la.Vector(ones), Math.random());
+                }
+            });
+        })
+    })
```

Comment:
serialization should also test non-initialized models (models that never saw any data)